### PR TITLE
Add job and config for knative-client release v0.2.0

### PR DESCRIPTION
**Pull request by navidshaikh**
As per title, branch release-v0.2.0 is in place.
Also updates the config for release-next branch.

### CODE REVIEW

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml [hunk 1]
```diff
@@ -22,8 +13,5 @@ tests:
     cluster_profile: aws
 resources:
   '*':
-    limits:
-      memory: 4Gi
     requests:
-      cpu: 100m
-      memory: 200Mi
```
Comment (petr-muller) [inline ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: If the resources are insufficient, make them larger, but I don't think this should be removed entirely.

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+binary_build_commands: make build
+tests:
+- as: e2e-aws
+  commands: "make test-e2e"
+  openshift_installer_src:
+    cluster_profile: aws
+resources:
+  '*':
+    requests:
+      memory: 2Gi
```
Comment (petr-muller) [inline ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml]: Same here

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+tag_specification:
```
Comment (petr-muller) [inline ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml]: This config builds no images, is that intentional?

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml (hunk 1 above)
Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: I am requesting 2Gi of memory (line no: 17).

You mean, restore the `limits` section ?
I removed the limits following the discussion on testplatform channel.
/cc @stevekuznetsov

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml (hunk 3 above)
Comment (navidshaikh) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-v0.2.0.yaml]: yes, it doesnt build any images, its a CLI.

File: ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply ci-operator/config/openshift/knative-client/openshift-knative-client-release-next.yaml]: Not having limits is good -- we don't want to deal with OOMkilled pods unless we have to

**Pull request merged**